pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -76,41 +80,49 @@ export const CryptoPolicyRow = () => {
                 <CryptoPolicyDialog close={() => setIsOpen(false)}
                                     currentCryptoPolicy={currentCryptoPolicy}
                                     setCurrentCryptoPolicy={setCurrentCryptoPolicy}
+                                    fipsEnabled={fipsEnabled}
                 />
                 }
             </td>
         </tr>
     );
 };
 
+const setPolicy = (policy, setError, setInProgress) => {
+    setInProgress(true);
+
+    const promises = [];
+    if (policy === "FIPS") {
+        promises.push(cockpit.spawn(["fips-mode-setup", "--enable"], { superuser: "require", err: "message" }));
+    } else {
+        promises.push(cockpit.spawn(["fips-mode-setup", "--disable"], { superuser: "require", err: "message" }).then(() =>
+            cockpit.spawn(["update-crypto-policies", "--set", policy], { superuser: "require", err: "message" })));
+    }
+
+    Promise.all(promises).then(() => {
+        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" }).catch(error => setError(error));
+    })
```

Comment:
I know it's not relevant here, as setPolicy() doesn't return a promise. But this is a rather subtle anti-pattern to get used to, also in case we'll ever use this in a different context. This code block as a whole will *not* wait for the inner spawn. Please drop the `{}` in the then arrow, so this is properly functional and chained.

(If this makes no sense to you, please do speak up! Happy to explain.)